frontend/src/modules/otherlocales/components/OtherLocales.js

```diff
@@ -37,41 +35,65 @@ export default class OtherLocales extends React.Component<Props> {
         </section>
     }
 
+    renderTranslations(
+        translation: OtherLocaleTranslation,
+        index: number,
+        lastPreferred: boolean,
+    ) {
+        return <Translation
+            entity={ this.props.entity }
+            isReadOnlyEditor={ this.props.isReadOnlyEditor }
+            translation={ translation }
+            parameters={ this.props.parameters }
+            updateEditorTranslation={ this.props.updateEditorTranslation }
+            lastPreferred = { lastPreferred }
+            key={ index }
+        />;
+    }
+
     render() {
-        const {
-            entity,
-            isReadOnlyEditor,
-            orderedOtherLocales,
-            preferredLocalesCount,
-            otherlocales,
-            parameters,
-            updateEditorTranslation,
-        } = this.props;
+        const { otherlocales } = this.props;
+
+        if (!otherlocales.translations) {
+            return null;
+        }
+
+        const translation = otherlocales.translations;
+        const preferredCount = translation.preferred.length;
 
         if (otherlocales.fetching) {
             return null;
         }
 
-        if (!otherlocales.translations.length) {
+        if (!translation.other.length && !translation.preferred.length) {
             return this.renderNoResults();
         }
 
-        return <section className="other-locales">
-            <ul>
-                { orderedOtherLocales.map((translation, index) => {
-                    let lastPreferred = (index === preferredLocalesCount - 1);
-
-                    return <Translation
-                        entity={ entity }
-                        isReadOnlyEditor={ isReadOnlyEditor }
-                        translation={ translation }
-                        parameters={ parameters }
-                        updateEditorTranslation={ updateEditorTranslation }
-                        lastPreferred={ lastPreferred }
-                        key={ index }
-                    />;
-                }) }
-            </ul>
+        return <section>
+            <section className="other-locales">
+                <ul>
+                    { translation.preferred.map((translation, index) => {
+                        const lastPreferred = (index === preferredCount -1);
```

Comment:
You can remove the `lastPreferred` logic from JS completely and use CSS to draw the green border bottom color on the last item in the first `<ul>`.